src/encoder.rs

```diff
@@ -89,7 +89,10 @@ impl<InstructionsType: Instructions> Encoder<InstructionsType> {
     pub fn encode_function(&mut self, function: &FunctionDefinition) -> FunctionVariables {
         for param in &function.parameters {
             let var = self.ssa_tracker.introduce_variable(param);
-            self.out(format!("(declare-const {} (_ BitVec 256))", var.name));
+            self.out(SMTExpression::DeclareConst(
+                var.name.as_smt(),
+                SMTIdentifier::bitvec(256),
```

Comment:
At first, I was a bit confused why SMTIdentifier is used here - maybe you could make it clearer that this is a sort?